### ISSUE TITLE
"On Screen" and "Confirm Only" PIN modes not working on Windows

### ISSUE THREAD

**Issue by sg495** [LedgerHQ/openpgp-card-app#31]
`scdaemon.conf` in my homedir is as described in the manual:

```
debug-level guru
debug-all
reader-port "Ledger Nano S 0"
enable-pinpad-varlen
```

A quick check with `gpgconf --list-options scdaemon` seems to show that the `enable-pinpad-varlen` option is indeed active:

```
Monitor:1:0:Options controlling the diagnostic output:0:0::::
verbose:12:0:verbose:0:0::::
quiet:0:0:be somewhat more quiet:0:0::::
Configuration:1:0:Options controlling the configuration:0:0::::
reader-port:8:0:connect to reader at port N:1:1:N:::""Ledger Nano S 0"
ctapi-driver:8:1:use NAME as ct-API driver:1:1:NAME:::
pcsc-driver:24:1:use NAME as PC/SC driver:1:1:NAME:"winscard.dll::
disable-pinpad:8:0:do not use a reader's pinpad:0:0::::
enable-pinpad-varlen:8:0:use variable length input for pinpad:0:0::::1
card-timeout:24:0:disconnect the card after N seconds of inactivity:3:3:N:0::
Debug:1:1:Options useful for debugging:0:0::::
debug-level:26:1:set the debugging level to LEVEL:1:1:LEVEL:"none::"guru
log-file:8:1:write a log to FILE:32:1:FILE:::
Security:1:0:Options controlling the security:0:0::::
deny-admin:8:0:deny the use of admin card commands:0:0::::
```

To my utter dismay, pgp continues to ask me to enter the pin in a Windows prompt...

**Comment by sg495** [comment]
Update: not working on Ubuntu 18.04 either.

**Comment by 1989gironimo** [comment]
After using verify it works for me, as you said in the other case.
I'm able to change the pin mode afterwards the way I want.

**Comment by Mansarde** [comment]
I have the same issue on Windows 10.
I.e. if I set "On Screen" or "Confirm Only" and then e.g. use the "name" command of gpg to change the name on the card, Windows still pops up a box to ask for the PIN instead of letting me enter it on the Ledger.

![image](https://user-images.githubusercontent.com/847687/43356860-cc422d2e-9278-11e8-812c-a049af72a0f6.png)

I also did a check via `gpgconf --list-options scdaemon` to verify that `enable-pinpad-varlen` was enabled:
```
enable-pinpad-varlen:8:0:use variable length input for pinpad:0:0::::1
```

GnuPG version used: 2.2.9
